gapis/api/vulkan/api/synchronization.api

```diff
@@ -583,4 +571,52 @@ cmd void vkCmdPipelineBarrier(
 
 extern void recordFenceSignal(VkFence fence)
 extern void recordFenceWait(VkFence fence)
-extern void recordFenceReset(VkFence fence)
\ No newline at end of file
+extern void recordFenceReset(VkFence fence)
+
+sub void processBarriers(VkPipelineStageFlags             srcStageMask        ,
+                         VkPipelineStageFlags             dstStageMask        ,
+                         map!(u32, VkMemoryBarrier)       memoryBarriers      ,
+                         map!(u32, VkBufferMemoryBarrier) bufferMemoryBarriers,
+                         map!(u32, VkImageMemoryBarrier)  imageMemoryBarriers ) {
+  // TODO: srcStageMask/dstStageMask
+
+  if len(memoryBarriers) > 0 {
+    // TODO: srcAccessMask/dstAccessMask
+    for _ , _ , img in Images {
```

Comment:
I am somewhat worried about the performance of this loop. It may be fine, as memory barriers are RELATIVELY few and far between, but we should double-check that.

Reply:
Would it make sense to move most of this code into a `@spy_disabled` subroutine? It looks like the image layout transition might be the only piece that is needed by gapii. Or are you also worried about `Mutate` performance hit? Do you know of any applications that might use more global memory barriers?